audio/writer-lame.c

```diff
@@ -0,0 +1,145 @@
+#include <assert.h>
+#include <lame/lame.h>
+#include <soundio/soundio.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "../util.h"
+#include "writer-lame.h"
+#include "writer.h"
+
+enum {
+	MP3_BUF_SIZE = 1UL << 22,
+};
+
+struct lame {
+	struct audio_writer writer;
+	FILE *outfile;
+	lame_t lflags;
+	unsigned char *buf;
+	size_t buf_size;
+};
+#define to_lame(ptr) container_of((ptr), struct lame, writer)
+
+static void
+lame_write(struct audio_writer *writer, enum SoundIoFormat fmt,
+	   char *data, int size, int bytes_per_frame)
+{
+	struct lame *lame;
+	char *left, *right;
+	int nsamples, blen = 0;
```

Comment:
Another stylistic thing, please stage declarations in order of line length from longest to shortest. In this example, `int nsamples...` and `struct lame ...` would be switched.